src/main/java/com/github/bordertech/config/DefaultConfiguration.java

```diff
@@ -1627,31 +1474,17 @@ public synchronized Object put(final Object aKey, final Object aValue) {
 					append = true;
 				}
 
-				if (append) {
+				if (append || INCLUDE_AFTER.equals(key)) {
 					String already = DefaultConfiguration.this.get(key);
 
 					// If there is no value already, strip off the leading comma, otherwise append.
 					value = (already != null ? already + "," + value : value);
 				}
 
-				DefaultConfiguration.this.load(key, value, location);
+				DefaultConfiguration.this.put(key, value, location);
 
 				return super.put(key, value);
 			}
 		}
-
-		@Override
-		public synchronized int hashCode() {
-			int hash = 5;
-			hash = 97 * hash + Objects.hashCode(this.location);
-			return hash;
-		}
-
-		@Override
-		public synchronized boolean equals(final Object obj) {
-			return obj instanceof IncludeProperties && Objects.equals(this.location, ((IncludeProperties) obj).location);
-		}
-
 	}
```

Comment:
Can we confirm it is OK to remove the hashCode and equals? Is the IncludeProperties being used as a key somewhere that relies on this?

Reply:
I can put it back in. Technically the IncludeProperties class exists to allow us to call the load() which subsequently calls the overridden put(Object, Object) to allow the Default Configuration to load properties into the backing HashMap in the order they are defined.

If you call load() and then loaded into the backing at the completion of load, the property definition order with change. This I believe is the only reason why IncludeProperties exists.

Reply:
After my long winded blurb, I have reinstated them. They are not used but there is no harm in keeping them there.